### PR TITLE
Adding btrfs udev rules might fix issue 947

### PR DESCRIPTION
**Pull request by johannbg**
This pull request changes...

## Checklist
- [ ] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #

### CODE REVIEW

File: modules.d/95udev-rules/module-setup.sh [hunk 1]
```diff
@@ -38,7 +38,9 @@ install() {
         60-pcmcia.rules \
         60-persistent-storage.rules \
         61-persistent-storage-edd.rules \
-        70-uaccess.rules \
+	64-btrfs-dm.rules \
```
Comment (danimo) [inline modules.d/95udev-rules/module-setup.sh:41]: Please fix indentation.

**Review by danimo — CHANGES_REQUESTED** [review]
See comment.

**Comment by haraldh** [comment]
looks good, although I would add them in `modules.d/90btrfs`

**Comment by johannbg** [comment]
@haraldh yeah I came to the same conclusion along with remove all this legacy cruff that is in that module. From my pov dracut should do a release then after that we need to seriously re-facture-it/rework it and remove all that legacy cruff like that old ( btrfs ) rules file etc. On top of that I took a wee bit closer look at that 64-btrfs-dm.rules I included and is being shipped with btrfs-progs in fedora and seems to be broken so that needs fixing in the *right* place as well so arguably we should not be including it. @haraldh @danimo thanx for reviewing this.

**Comment by danimo** [comment]
@haraldh fine with merging as-is?

**Comment by haraldh** [comment]
> @haraldh fine with merging as-is?

@danimo yes

**Comment by johannbg** [comment]
@danimo @haraldh Note that 64-btrfs-dm.rules ( which come from btrfs-progs ) and I included are outdated and probably break with systemd 247+ so by adding it we might be signing ourselves up for more issues.

**Comment by johannbg** [comment]
.

**Comment by cmurf** [comment]
I don't see how /usr/lib/udev/rules.d/64-btrfs-dm.rules would fix #947. What's missing from the initrd in that bug is /usr/lib/udev/rules.d/64-btrfs.rules to make sure systemd waits for all btrfs devices to be ready before trying to mount. 

/usr/lib/udev/rules.d/64-btrfs.rules is part of systemd

/usr/lib/udev/rules.d/64-btrfs-dm.rules is part of btrfs-progs, if that's somehow stale or out of date, it needs an issue filed with [btrfsprogs](https://github.com/kdave/btrfs-progs/issues)

**Comment by johannbg** [comment]
Let's start by getting the udev rules into the reporters initramfs before claiming that it's not the problem and I'm very well aware of where I could file reports against btrfs-prog and those rules are broken ( as well as bunch of other udev rules ) but thanks for the info. We will look at this properly when we have the time...

**Comment by cmurf** [comment]
The [problem is resolved](https://ask.fedoraproject.org/t/btrfs-volume-mount-fails-at-boot-but-works-once-system-is-up/9593/23?u=chrismurphy) when `dracut -f --add btrfs` is used, which results in usr/lib/udev/rules.d/64-btrfs.rules being included in the initramfs. Whereas /usr/lib/udev/rules.d/64-btrfs-dm.rules is absent. In the original report, as well as my testing, dm isn't being used for anything.

**Pull request closed without merging**